apisix/core/ai.lua

```diff
@@ -17,6 +17,24 @@
 local require         = require
 local core            = require("apisix.core")
 local ipairs          = ipairs
+local pcall           = pcall
+local loadstring      = loadstring
+local encode_base64   = ngx.encode_base64
+
+local get_cache_key_func
+local get_cache_key_func_def_render
+
+local get_cache_key_func_def = [[
+return function(ctx)
+    return ctx.var.uri
```

Comment:
We can make it a little faster, by using `local var = ctx.var` to reduce table lookup.